src/keyring/ed25519/ledgertm.rs

```diff
@@ -0,0 +1,36 @@
+//! Ledger Tendermint signer
+
+use signatory::PublicKeyed;
+use signatory_ledger_tm::{self, Ed25519LedgerTmAppSigner};
+
+use crate::{
+    config::provider::ledgertm::LedgerTendermintConfig,
+    error::{KmsError, KmsErrorKind::*},
+    keyring::{ed25519::Signer, KeyRing},
+};
+
+pub const LEDGER_TM_PROVIDER_LABEL: &str = "ledgertm";
+pub const LEDGER_TM_ID: &str = "ledgertm";
+
+/// Create Ledger Tendermint signer object from the given configuration
+pub fn init(
+    keyring: &mut KeyRing,
+    ledgertm_configs: &[LedgerTendermintConfig],
+) -> Result<(), KmsError> {
+    if ledgertm_configs.is_empty() {
+        return Ok(());
+    }
+
+    if ledgertm_configs.len() != 1 {
+        fail!(
+            ConfigError,
+            "expected one [providers.ledgertm] in config, found: {}",
+            ledgertm_configs.len()
+        );
+    }
+    let provider = Box::new(Ed25519LedgerTmAppSigner::connect()?);
+    let pk = provider.public_key()?;
+    let signer = Signer::new(LEDGER_TM_PROVIDER_LABEL, LEDGER_TM_ID.to_string(), provider);
```

Comment:
@jlenil Will there be an equivalent to a key_id we can put into the config? e.g. for yubihsm we can have sth like `keys = [{ id = "gaia-9000", key = 1 }]`. I guess this makes sense here, too?
I'll add a TODO here and merge for now.

Reply:
The validator app needs to rely on Ledger's crypto API for Ed25519 so in practice there is a Bip32 derivation path. At the moment, this is not exposed in the API but it would be actually possible to have different keys to allow for something like that. Actually, it could be even possible to have both secp256k1 and ed25519 in the same device.

Should we open an issue for this new feature?

Reply:
Yes, please that would be awesome! Thanks :)

Reply:
Created issues for this:
https://github.com/tendermint/kms/issues/177
https://github.com/cosmos/ledger-cosmos/issues/108